src/util/statistics_registry.cpp

```diff
@@ -240,6 +246,20 @@ RegisterStatistic::~RegisterStatistic() {
   d_reg->unregisterStat(d_stat);
 }
 
+template <>
+void ReferenceStat<std::string>::safeFlushInformation(int fd) const {
+  if (__CVC4_USE_STATISTICS) {
+    safe_print<const std::string*>(fd, d_data);
+  }
+}
+
+template <>
+void BackedStat<std::string>::safeFlushInformation(int fd) const {
```

Comment:
These specializations seem a little off. Why do we need them and could we avoid having them by subclassing differently?